pandas/core/dtypes/cast.py

```diff
@@ -1328,6 +1328,32 @@ def common_dtype_categorical_compat(
     return dtype
 
 
+def np_find_common_type(*dtypes: np.dtype) -> np.dtype:
+    """
+    np.find_common_type implementation pre-1.25 deprecation using np.result_type
+    https://github.com/pandas-dev/pandas/pull/49569#issuecomment-1308300065
+
+    Parameters
+    ----------
+    dtypes : np.dtypes
+
+    Returns
+    -------
+    np.dtype
+    """
+    try:
+        common_dtype = np.result_type(*dtypes)
+        if common_dtype.kind in "mMSU":
+            # NumPy promotion currently (1.25) misbehaves for for times and strings,
```

Comment:
is there some numpy version at which we can remove this check?

Reply:
Not sure cc @seberg any thoughts?

Reply:
I have a PR that would fix it, so I hope next release this will happen.  That was the PR which get stuck a bit on the discussion of what `==` should do.